### PR TITLE
jibri: add jibri service

### PR DESCRIPTION
**Pull request by netaskd**
Add JItsi BRoadcasting Infrastructure in docker infrastructure.

### CODE REVIEW

**Comment by saghul** [comment]
Ok, now that I finally got the release out of the way, let's try to get Jibri in. Can you please rebase? Also, we promoted jibri to the stable Debian repo, so no need to switch repos to install it!

**Comment by netaskd** [comment]
@saghul I've done rebase. Please check it.

File: jibri/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+FROM jitsi/base-java
+
+ARG JITSI_RELEASE=unstable
```
Comment (saghul) [inline jibri/Dockerfile]: this can go now

Comment (netaskd) [reply jibri/Dockerfile]: done

File: README.md [hunk 2]
```diff
@@ -49,6 +53,7 @@ A Jitsi Meet installation can be broken down into the following components:
 * A conference focus component
 * A video router (could be more than one)
 * A SIP gateway for audio calls
+* A BRoadcasting Infrastructure for recording or streaming a conference.
```
Comment (sapkra) [inline README.md]:
```suggestion
* A Broadcasting Infrastructure for recording or streaming a conference.
```

Comment (netaskd) [reply README.md]: thanks, fixed in 4f76c40

**Comment by vincentpalita** [comment]
any chance it gets merged soon :) ?

**Comment by chrisstpierre** [comment]
This doesn't work for me. Is that why this hasn't been accepted yet?  It's having some issues with connecting to the xmpp server. Does anyone have some insight? Otherwise it seems like the PR could be accepted

```
18:27:10
2019-10-29 14:27:10.995 INFO: [34] org.jitsi.xmpp.mucclient.MucClient.log() [MucClient id=xmpp.meet.jitsi hostname=xmpp.meet.jitsi] closed on error:

18:27:10
org.jivesoftware.smack.XMPPException$StreamErrorException: connection-timeout You can read more about the meaning of this stream error at http://xmpp.org/rfcs/rfc6120.html#streams-error-conditions
org.jivesoftware.smack.XMPPException$StreamErrorException: connection-timeout You can read more about the meaning of this stream error at http://xmpp.org/rfcs/rfc6120.html#streams-error-conditions

18:27:10
<stream:error><connection-timeout xmlns='urn:ietf:params:xml:ns:xmpp-streams'/></stream:error>

18:27:10
at org.jivesoftware.smack.tcp.XMPPTCPConnection$PacketReader.parsePackets(XMPPTCPConnection.java:1064)

18:27:10
at org.jivesoftware.smack.tcp.XMPPTCPConnection$PacketReader.access$300(XMPPTCPConnection.java:1000)

18:27:10
at org.jivesoftware.smack.tcp.XMPPTCPConnection$PacketReader$1.run(XMPPTCPConnection.java:1016)

18:27:10
at java.lang.Thread.run(Thread.java:748)
```

**Comment by chrisstpierre** [comment]
It wouldn't work until I manually registered the Jibri user like so:
```
prosodyctl register jibri auth.yourdomain.com jibriauthpass
prosodyctl register recorder recorder.yourdomain.com jibrirecorderpass
```

Maybe that will help somebody

**Comment by netaskd** [comment]
@chrisstpierre Do you have set variables `JIBRI_XMPP_USER` /`JIBRI_XMPP_PASSWORD` and `JIBRI_RECORDER_USER`/`JIBRI_RECORDER_PASSWORD`?
Prosodyctl register logic exists in code:
https://github.com/jitsi/docker-jitsi-meet/blob/1566bec496f9e540127ba4045bf64d0dbe68ed3b/prosody/rootfs/etc/cont-init.d/10-config#L40
https://github.com/jitsi/docker-jitsi-meet/blob/1566bec496f9e540127ba4045bf64d0dbe68ed3b/prosody/rootfs/etc/cont-init.d/10-config#L44

**Comment by chrisstpierre** [comment]
Yes I did notice the prosodyctl logic, & yes `JIBRI_XMPP_USER` & `JIBRI_XMPP_PASSWORD` was set. I'm not 100% clear on the issue but I thought I'd share anyways incase it helps

**Comment by saghul** [comment]
Alright, I finally got to merging this! ffa017b 🚀 🚀 🚀 

Huge kudos to @netaskd, you really did a nice job here. I made a few minor changes (missing env vars in docker-compose.yml, also duplicated one), README tweaks and enabled the live streaming option too.

I'll be making a new release of all the images, including jibri.

**Comment by netaskd** [comment]
@saghul,
Great! TURN on the way... ;)

**Comment by Bellusterra** [comment]
> Alright, I finally got to merging this! [ffa017b](https://github.com/jitsi/docker-jitsi-meet/commit/ffa017b74d57ba2580fe6647f8de06b012aa4ef2) rocket rocket rocket
> 
> Huge kudos to @netaskd, you really did a nice job here. I made a few minor changes (missing env vars in docker-compose.yml, also duplicated one), README tweaks and enabled the live streaming option too.
> 
> I'll be making a new release of all the images, including jibri.

@saghul I think the duplicate is still in the README.md file I see the variable JIBRI_RECORDER_PASSWORD twice, instead of the variable JIBRI_XMPP_PASSWORD for the second one.

**Comment by Bellusterra** [comment]
@saghul do multiple jibri containers need to be run in separate VM’s or can they all be run on one host machine? I have changed the loopback settings and the nicknames for the individual containers but with no luck. I had both the complete jitsi docker setup working and recording in the meet.jitsi network and a separate bare metal install setup working and recording and at the same time on the same host, but trying to run docker jibri on the host machine directly in “host“ mode I get an xorg error.

**Comment by saghul** [comment]
Generally it’s simpler to run separate VMs since there will be less fiddling with the config, and the resource usage is now low anyway, so if you choose the right size VM you’ll be utilizing it to its fullest.

**Pull request closed without merging**